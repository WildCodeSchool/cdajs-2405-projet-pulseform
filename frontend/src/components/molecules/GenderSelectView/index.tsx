import { useEffect, useState } from "react";
import { useTranslation } from "react-i18next";
import { EffectCoverflow } from "swiper/modules";
import { Swiper, SwiperSlide } from "swiper/react";

import BasicButton from "@components/atoms/BasicButton";
import BodyStepQuestions from "@components/molecules/BodyStepQuestions";
import ExitSignUpStep from "@components/molecules/ExitSignUpStep";
import FitnessLevelSelectView from "@components/molecules/FitnessLevelSelectView";

import "./GenderSelectView.scss";
import "swiper/css";
import "swiper/css/effect-coverflow";
import blueCross from "@assets/icons/blue-cross.svg";
import girl2 from "@assets/images/girl2.svg";
import men from "@assets/images/men.svg";
import noAnswerSex from "@assets/images/no-answer-sex.svg";
import otherSex from "@assets/images/other-sex.svg";

function GenderSelectView() {
  const [isExit, setIsExit] = useState(false);
  const [spaceBetween, setSpaceBetween] = useState(130);
  const [currentSlideId, setCurrentSlideId] = useState<number | null>(null);
  const [isNextStep, setIsNextStep] = useState(false);

  const { t } = useTranslation();

  function handleExit() {
    setIsExit(true);
  }

  function handleNextStep() {
    setIsNextStep(true);
  }

  const slides = [
    { id: 1, content: "GIRL", src: girl2 },
    { id: 2, content: "MEN", src: men },
    { id: 3, content: "OTHER", src: otherSex },
    { id: 4, content: "NOT_WISH_TO_ANSWER", src: noAnswerSex },
  ];

  type Swiper = {
    activeIndex: number;
  };

  const handleSlideChange = (swiper: Swiper): void => {
    const activeIndex = swiper.activeIndex; // Index de la diapositive visible au centre
    const activeSlideId = slides[activeIndex]?.id || null; // Récupère l'ID correspondant
    setCurrentSlideId(activeSlideId);
  };

  useEffect(() => {
    function handleResize() {
      const screenWidth = window.innerWidth;

      if (screenWidth > 950) {
        setSpaceBetween(-200);
      } else if (screenWidth > 700) {
        setSpaceBetween(60);
      } else {
        setSpaceBetween(130);
      }
    }

    handleResize();
    window.addEventListener("resize", handleResize);

    return () => window.removeEventListener("resize", handleResize);
  }, []);

  return (
    <>
      {!isExit && !isNextStep && (
        <>
          <section className="gender-select-view">
            <button
              className="gender-select-view__exit"
              type="button"
              onClick={handleExit}
            >
              <img src={blueCross} alt="Close" />
            </button>
            <p className="gender-select-view__question">{t("YOUR_ARE")}</p>
            <Swiper
              className="gender-select-view__swiper-container"
              effect="coverflow"
              grabCursor={true}
              centeredSlides={true}
              slidesPerView={2}
              spaceBetween={spaceBetween}
              onSlideChange={handleSlideChange}
              style={{ height: "70vh" }}
              coverflowEffect={{
                rotate: 30,
                stretch: 10,
                depth: 150,
                modifier: 1,
                slideShadows: false,
              }}
              modules={[EffectCoverflow]}
            >
              {slides.map((slide) => (
                <SwiperSlide key={slide.id} style={{ height: "100%" }}>
                  <img
                    className="gender-select-view__swiper-container__img"
                    src={slide.src}
                    alt={slide.content}
                    style={{ height: "100%", objectFit: "contain" }}
                  />
                  <p className="gender-select-view__swiper-container__content">
                    {t(slide.content)}
                  </p>
                </SwiperSlide>
              ))}
            </Swiper>
            <BasicButton
              className="gender-select-view__btn"
              type="button"
              onClick={handleNextStep}
            >
              {t("NEXT")}
            </BasicButton>
            <p>Diapositive active : {currentSlideId || 1}</p>
          </section>
          <section className="gender-select-view-d">
            <BodyStepQuestions
              questionLabel={t("YOUR_ARE")}
              ctaExit={handleExit}
            >
              <form className="gender-select-view-d__form" action="">
                <BasicButton
                  typeButton="white"
                  type="button"
                  onClick={() => setCurrentSlideId(1)}
                  hasFocus
                >
                  {t("GIRL")}
                </BasicButton>
                <BasicButton
                  typeButton="white"
                  type="button"
                  onClick={() => setCurrentSlideId(2)}
                  hasFocus
                >
                  {t("MEN")}
                </BasicButton>
                <BasicButton
                  typeButton="white"
                  onClick={() => setCurrentSlideId(3)}
                  hasFocus
                >
                  {t("OTHER")}
                </BasicButton>
                <BasicButton
                  typeButton="white"
                  type="button"
                  onClick={() => setCurrentSlideId(4)}
                  hasFocus
                >
                  {t("NOT_WISH_TO_ANSWER")}
                </BasicButton>
                <BasicButton
                  type="submit"
                  onClick={handleNextStep}
                  disabled={currentSlideId === null}
                >
                  {t("NEXT")}
                </BasicButton>
                <p style={{ color: "white" }}>
                  Diapositive active : {currentSlideId || 1}
                </p>
              </form>
            </BodyStepQuestions>
          </section>
        </>
      )}
      {isExit && <ExitSignUpStep />}
      {!isExit && isNextStep && <FitnessLevelSelectView />}
    </>
  );
}

export default GenderSelectView;
