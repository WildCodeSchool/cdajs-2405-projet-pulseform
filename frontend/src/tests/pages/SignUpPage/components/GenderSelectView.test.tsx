import GenderSelectView from "@pages/SignUpPage/components/GenderSelectView";
import { render } from "@testing-library/react";

describe("GenderSelectView", () => {
  it("renders the GenderSelectView component", () => {
    const tree = render(<GenderSelectView />);
    expect(tree).toMatchSnapshot();
  });
});
