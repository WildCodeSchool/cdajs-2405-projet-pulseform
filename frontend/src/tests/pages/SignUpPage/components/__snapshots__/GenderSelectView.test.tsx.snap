// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GenderSelectView > renders the GenderSelectView component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="gender-select-view"
      >
        <button
          class="gender-select-view__exit"
          type="button"
        >
          <img
            alt="Close"
            src="/src/assets/icons/blue-cross.svg"
          />
        </button>
        <p
          class="gender-select-view__question"
        >
          YOUR_ARE
        </p>
        <div
          class="swiper swiper-coverflow swiper-3d swiper-initialized swiper-horizontal swiper-watch-progress gender-select-view__swiper-container"
          style="height: 70vh;"
        >
          <div
            class="swiper-wrapper"
            style="cursor: grab; transition-duration: 0ms; transition-delay: 0ms; transform: translate3d(NaNpx, 0px, 0px);"
          >
            <div
              class="swiper-slide"
              style="height: 100%;"
            >
              <img
                alt="GIRL"
                class="gender-select-view__swiper-container__img"
                src="/src/assets/images/girl2.svg"
                style="height: 100%; object-fit: contain;"
              />
              <p
                class="gender-select-view__swiper-container__content"
              >
                GIRL
              </p>
            </div>
            <div
              class="swiper-slide"
              style="height: 100%;"
            >
              <img
                alt="MEN"
                class="gender-select-view__swiper-container__img"
                src="/src/assets/images/men.svg"
                style="height: 100%; object-fit: contain;"
              />
              <p
                class="gender-select-view__swiper-container__content"
              >
                MEN
              </p>
            </div>
            <div
              class="swiper-slide"
              style="height: 100%;"
            >
              <img
                alt="OTHER"
                class="gender-select-view__swiper-container__img"
                src="/src/assets/images/other-sex.svg"
                style="height: 100%; object-fit: contain;"
              />
              <p
                class="gender-select-view__swiper-container__content"
              >
                OTHER
              </p>
            </div>
            <div
              class="swiper-slide"
              style="height: 100%;"
            >
              <img
                alt="NOT_WISH_TO_ANSWER"
                class="gender-select-view__swiper-container__img"
                src="/src/assets/images/no-answer-sex.svg"
                style="height: 100%; object-fit: contain;"
              />
              <p
                class="gender-select-view__swiper-container__content"
              >
                NOT_WISH_TO_ANSWER
              </p>
            </div>
          </div>
        </div>
        <button
          class="basic-button gender-select-view__btn basic-button--basic"
          type="button"
        >
          NEXT
        </button>
        <p>
          Diapositive active : 
          1
        </p>
      </section>
      <section
        class="gender-select-view-d"
      >
        <div
          class="body-step-questions"
        >
          <div
            class="body-step-questions__header-d"
          >
            <div
              class="body-step-questions__header-d__logo"
            >
              <div
                class="little-logo little-logo--desktop"
              >
                <img
                  alt="Logo Pulse Form"
                  class="little-logo__img"
                  src="/src/assets/icons/mobile-logo.svg"
                />
                <p
                  class="little-logo__title little-logo__title--desktop"
                >
                  Pulse Form
                </p>
              </div>
            </div>
            <div
              class="body-step-questions__header-d__container"
            >
              <button
                class="body-step-questions__header-d__container__btn"
                type="button"
              >
                TEST_PROGRAM
              </button>
            </div>
          </div>
          <button
            class="body-step-questions__exit"
            type="button"
          >
            <img
              alt="Close"
              src="/src/assets/icons/blue-cross.svg"
            />
          </button>
          <h2
            class="body-step-questions__question"
          >
            YOUR_ARE
          </h2>
          <img
            alt="blob-girl"
            class="body-step-questions__blob-girl"
            src="/src/assets/images/blob-girl.svg"
          />
          <div
            class="body-step-questions__container"
          >
            <form
              action=""
              class="fitness-level-select-view__container__form"
            >
              <button
                class="basic-button fitness-level-select-view__container__form__btn basic-button--white basic-button--white--focus"
                type="button"
              >
                GIRL
              </button>
              <button
                class="basic-button fitness-level-select-view__container__form__btn basic-button--white basic-button--white--focus"
                type="button"
              >
                MEN
              </button>
              <button
                class="basic-button fitness-level-select-view__container__form__btn basic-button--white basic-button--white--focus"
                type="button"
              >
                OTHER
              </button>
              <button
                class="basic-button fitness-level-select-view__container__form__btn basic-button--white basic-button--white--focus"
                type="button"
              >
                NOT_WISH_TO_ANSWER
              </button>
              <button
                class="basic-button fitness-level-select-view__container__form__btn basic-button--basic"
                disabled=""
                type="submit"
              >
                NEXT
              </button>
              <p
                style="color: white;"
              >
                Diapositive active : 
                1
              </p>
            </form>
          </div>
          <div
            class="body-step-questions__container-d"
          >
            <div
              class="body-step-questions__container-d__question"
            >
              <img
                alt="sport"
                class="body-step-questions__container-d__question__img"
                src="/src/assets/images/step-img.svg"
              />
              <div
                class="body-step-questions__container-d__question__container"
              >
                <p
                  class="body-step-questions__container-d__question__container__label"
                >
                  MOTIVATIONAL_LABEL
                </p>
              </div>
            </div>
            <div
              class="body-step-questions__container-d__children"
            >
              <p
                class="body-step-questions__container-d__children__label"
              >
                YOUR_ARE
              </p>
              <form
                action=""
                class="fitness-level-select-view__container__form"
              >
                <button
                  class="basic-button fitness-level-select-view__container__form__btn basic-button--white basic-button--white--focus"
                  type="button"
                >
                  GIRL
                </button>
                <button
                  class="basic-button fitness-level-select-view__container__form__btn basic-button--white basic-button--white--focus"
                  type="button"
                >
                  MEN
                </button>
                <button
                  class="basic-button fitness-level-select-view__container__form__btn basic-button--white basic-button--white--focus"
                  type="button"
                >
                  OTHER
                </button>
                <button
                  class="basic-button fitness-level-select-view__container__form__btn basic-button--white basic-button--white--focus"
                  type="button"
                >
                  NOT_WISH_TO_ANSWER
                </button>
                <button
                  class="basic-button fitness-level-select-view__container__form__btn basic-button--basic"
                  disabled=""
                  type="submit"
                >
                  NEXT
                </button>
                <p
                  style="color: white;"
                >
                  Diapositive active : 
                  1
                </p>
              </form>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="gender-select-view"
    >
      <button
        class="gender-select-view__exit"
        type="button"
      >
        <img
          alt="Close"
          src="/src/assets/icons/blue-cross.svg"
        />
      </button>
      <p
        class="gender-select-view__question"
      >
        YOUR_ARE
      </p>
      <div
        class="swiper swiper-coverflow swiper-3d swiper-initialized swiper-horizontal swiper-watch-progress gender-select-view__swiper-container"
        style="height: 70vh;"
      >
        <div
          class="swiper-wrapper"
          style="cursor: grab; transition-duration: 0ms; transition-delay: 0ms; transform: translate3d(NaNpx, 0px, 0px);"
        >
          <div
            class="swiper-slide"
            style="height: 100%;"
          >
            <img
              alt="GIRL"
              class="gender-select-view__swiper-container__img"
              src="/src/assets/images/girl2.svg"
              style="height: 100%; object-fit: contain;"
            />
            <p
              class="gender-select-view__swiper-container__content"
            >
              GIRL
            </p>
          </div>
          <div
            class="swiper-slide"
            style="height: 100%;"
          >
            <img
              alt="MEN"
              class="gender-select-view__swiper-container__img"
              src="/src/assets/images/men.svg"
              style="height: 100%; object-fit: contain;"
            />
            <p
              class="gender-select-view__swiper-container__content"
            >
              MEN
            </p>
          </div>
          <div
            class="swiper-slide"
            style="height: 100%;"
          >
            <img
              alt="OTHER"
              class="gender-select-view__swiper-container__img"
              src="/src/assets/images/other-sex.svg"
              style="height: 100%; object-fit: contain;"
            />
            <p
              class="gender-select-view__swiper-container__content"
            >
              OTHER
            </p>
          </div>
          <div
            class="swiper-slide"
            style="height: 100%;"
          >
            <img
              alt="NOT_WISH_TO_ANSWER"
              class="gender-select-view__swiper-container__img"
              src="/src/assets/images/no-answer-sex.svg"
              style="height: 100%; object-fit: contain;"
            />
            <p
              class="gender-select-view__swiper-container__content"
            >
              NOT_WISH_TO_ANSWER
            </p>
          </div>
        </div>
      </div>
      <button
        class="basic-button gender-select-view__btn basic-button--basic"
        type="button"
      >
        NEXT
      </button>
      <p>
        Diapositive active : 
        1
      </p>
    </section>
    <section
      class="gender-select-view-d"
    >
      <div
        class="body-step-questions"
      >
        <div
          class="body-step-questions__header-d"
        >
          <div
            class="body-step-questions__header-d__logo"
          >
            <div
              class="little-logo little-logo--desktop"
            >
              <img
                alt="Logo Pulse Form"
                class="little-logo__img"
                src="/src/assets/icons/mobile-logo.svg"
              />
              <p
                class="little-logo__title little-logo__title--desktop"
              >
                Pulse Form
              </p>
            </div>
          </div>
          <div
            class="body-step-questions__header-d__container"
          >
            <button
              class="body-step-questions__header-d__container__btn"
              type="button"
            >
              TEST_PROGRAM
            </button>
          </div>
        </div>
        <button
          class="body-step-questions__exit"
          type="button"
        >
          <img
            alt="Close"
            src="/src/assets/icons/blue-cross.svg"
          />
        </button>
        <h2
          class="body-step-questions__question"
        >
          YOUR_ARE
        </h2>
        <img
          alt="blob-girl"
          class="body-step-questions__blob-girl"
          src="/src/assets/images/blob-girl.svg"
        />
        <div
          class="body-step-questions__container"
        >
          <form
            action=""
            class="fitness-level-select-view__container__form"
          >
            <button
              class="basic-button fitness-level-select-view__container__form__btn basic-button--white basic-button--white--focus"
              type="button"
            >
              GIRL
            </button>
            <button
              class="basic-button fitness-level-select-view__container__form__btn basic-button--white basic-button--white--focus"
              type="button"
            >
              MEN
            </button>
            <button
              class="basic-button fitness-level-select-view__container__form__btn basic-button--white basic-button--white--focus"
              type="button"
            >
              OTHER
            </button>
            <button
              class="basic-button fitness-level-select-view__container__form__btn basic-button--white basic-button--white--focus"
              type="button"
            >
              NOT_WISH_TO_ANSWER
            </button>
            <button
              class="basic-button fitness-level-select-view__container__form__btn basic-button--basic"
              disabled=""
              type="submit"
            >
              NEXT
            </button>
            <p
              style="color: white;"
            >
              Diapositive active : 
              1
            </p>
          </form>
        </div>
        <div
          class="body-step-questions__container-d"
        >
          <div
            class="body-step-questions__container-d__question"
          >
            <img
              alt="sport"
              class="body-step-questions__container-d__question__img"
              src="/src/assets/images/step-img.svg"
            />
            <div
              class="body-step-questions__container-d__question__container"
            >
              <p
                class="body-step-questions__container-d__question__container__label"
              >
                MOTIVATIONAL_LABEL
              </p>
            </div>
          </div>
          <div
            class="body-step-questions__container-d__children"
          >
            <p
              class="body-step-questions__container-d__children__label"
            >
              YOUR_ARE
            </p>
            <form
              action=""
              class="fitness-level-select-view__container__form"
            >
              <button
                class="basic-button fitness-level-select-view__container__form__btn basic-button--white basic-button--white--focus"
                type="button"
              >
                GIRL
              </button>
              <button
                class="basic-button fitness-level-select-view__container__form__btn basic-button--white basic-button--white--focus"
                type="button"
              >
                MEN
              </button>
              <button
                class="basic-button fitness-level-select-view__container__form__btn basic-button--white basic-button--white--focus"
                type="button"
              >
                OTHER
              </button>
              <button
                class="basic-button fitness-level-select-view__container__form__btn basic-button--white basic-button--white--focus"
                type="button"
              >
                NOT_WISH_TO_ANSWER
              </button>
              <button
                class="basic-button fitness-level-select-view__container__form__btn basic-button--basic"
                disabled=""
                type="submit"
              >
                NEXT
              </button>
              <p
                style="color: white;"
              >
                Diapositive active : 
                1
              </p>
            </form>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
